metamorph/src/test/java/org/metafacture/metamorph/TestMetamorphBasics.java

```diff
@@ -152,13 +238,118 @@ public void shouldHandleUnmatchedLiteralsAndEntitiesInElseNestedSource() {
+                }
+        );
+    }
+
+    @Test
+    public void shouldHandlePartiallyUnmatchedLiteralsAndEntitiesInElseNestedSource() {
+        assertMorph(receiver,
+                "<rules>" +
+                "  <entity name='USA1'>" +
+                "    <data source='USA1.Sylt' name='Hawaii' />" +
+                "  </entity>" +
+                "  <entity name='USA2'>" +
+                "    <data source='USA2.Sylt' name='Hawaii' />" +
+                "  </entity>" +
+                "  <entity name='USA3'>" +
+                "    <data source='USA3.Sylt' name='Hawaii' />" +
+                "  </entity>" +
+                "  <entity name='USA4'>" +
+                "    <data source='USA4.Sylt' name='Hawaii' />" +
+                "  </entity>" +
+                "  <data source='_elseNested' />" +
+                "</rules>",
+                i -> {
+                    i.startRecord("1");
+                    i.literal("Shikotan", "Aekap");
+                    i.startEntity("Germany");
+                    i.literal("Langeoog", "Moin");
+                    i.literal("Baltrum", "Moin Moin");
+                    i.endEntity();
+                    i.startEntity("USA1");
+                    i.literal("Sylt", "Aloha");
+                    i.endEntity();
+                    i.startEntity("USA2");
+                    i.literal("Sylt", "Aloha");
+                    i.literal("Langeoog", "Moin");
+                    i.literal("Baltrum", "Moin Moin");
+                    i.endEntity();
+                    i.startEntity("USA3");
+                    i.literal("Langeoog", "Moin");
+                    i.literal("Sylt", "Aloha");
+                    i.literal("Baltrum", "Moin Moin");
+                    i.endEntity();
+                    i.startEntity("USA4");
+                    i.literal("Langeoog", "Moin");
+                    i.literal("Baltrum", "Moin Moin");
+                    i.literal("Sylt", "Aloha");
+                    i.endEntity();
+                    i.endRecord();
+                },
+                (o, f) -> {
+                    final boolean coordinatesWithEntity = false;
+                    final boolean separatesFromEntity = false;
+                    o.get().startRecord("1");
+                    o.get().literal("Shikotan", "Aekap");
                     o.get().startEntity("Germany");
+                    o.get().literal("Langeoog", "Moin");
                     o.get().literal("Baltrum", "Moin Moin");
                     o.get().endEntity();
-                    o.get().startEntity("USA");
+                    o.get().startEntity("USA1");
+                    o.get().literal("Hawaii", "Aloha");
+                    o.get().endEntity();
+                    o.get().startEntity("USA2");
+                    o.get().literal("Hawaii", "Aloha");
+                    if (!coordinatesWithEntity) {
+                    o.get().endEntity();
+                    o.get().startEntity("USA2");
```

Comment:
indentation?

Reply:
If we decide to leave it in, yes. But for the time being it was only intended to illustrate the issue (and the different results, depending on the potential resolution).